test/ldap_test.rb

```diff
@@ -117,12 +118,79 @@ def test_search_strategy_misconfigured_to_unrecognized_strategy_falls_back_to_de
   def test_capabilities
     assert_kind_of Net::LDAP::Entry, @ldap.capabilities
   end
+
+  def test_global_catalog_connection_is_null_if_not_active_directory
+    @ldap.expects(:active_directory_capability?).returns(false)
+    assert_equal nil, @ldap.global_catalog_connection
+  end
+
+  def test_global_catalog_connection_is_null_if_not_active_directory
+    @ldap.expects(:active_directory_capability?).returns(true)
+    refute_nil @ldap.global_catalog_connection, "Expected Global Catalog to not be nil"
+  end
+
+  def test_global_catalog_returns_empty_array_for_no_results
+    @ldap.expects(:active_directory_capability?).returns(true)
+    mock_global_catalog_connection = Object.new
+    mock_global_catalog_connection.expects(:search).returns(nil)
+    Net::LDAP.expects(:new).returns(mock_global_catalog_connection)
+    results = @ldap.global_catalog_search({})
+    assert_equal [], results
+  end
+
+  def test_global_catalog_returns_array_of_results
+    @ldap.expects(:active_directory_capability?).returns(true)
+    mock_global_catalog_connection = Object.new
+    stub_entry = Object.new
+    mock_global_catalog_connection.expects(:search).returns(stub_entry)
+    Net::LDAP.expects(:new).returns(mock_global_catalog_connection)
+    results = @ldap.global_catalog_search({})
+    assert_equal [stub_entry], results
+  end
+
+  def test_global_catalog_default_settings
+    @ldap.expects(:active_directory_capability?).returns(true)
+    global_catalog = @ldap.global_catalog_connection
+    instrumentation_service = global_catalog.instance_variable_get(:@instrumentation_service)
+
+    auth = global_catalog.instance_variable_get(:@auth)
+    assert_equal :simple, auth[:method]
+    assert_equal "localhost", global_catalog.host
+    assert_equal 3268, global_catalog.port
+    assert_equal "MockInstrumentationService", instrumentation_service.class.name
+  end
+
+  module GitHubLdapUnauthenticatedTestCases
+    def test_global_catalog_unauthenticated_default_settings
+      @ldap.expects(:active_directory_capability?).returns(true)
+      global_catalog = @ldap.global_catalog_connection
+      # this is ugly, but currently the only way to test Net::LDAP#auth values
+      auth = global_catalog.instance_variable_get(:@auth)
+
+      assert_equal nil, auth[:password]
+      assert_equal nil, auth[:username]
+    end
+  end
+
+  module GitHubLdapAuthenticatedTestCases
+    def test_global_catalog_authenticated_default_settings
+      @ldap.expects(:active_directory_capability?).returns(true)
+      global_catalog = @ldap.global_catalog_connection
+      # this is ugly, but currently the only way to test Net::LDAP#auth values
+      auth = global_catalog.instance_variable_get(:@auth)
+
+      assert_equal "passworD1", auth[:password]
+      assert_equal "uid=admin,dc=github,dc=com", auth[:username]
```

Comment:
What's the intention behind these tests? Are the settings manipulated somehow or changed, or are they the same as the ones we pass in to the initializer? I'm confused what failure we're trying to catch here.